### PR TITLE
MAINT: Handle warnings in tutorials

### LINKED ISSUE #154: Fail CI for warnings
Ideally we should handle warnings raised during the tutorials, and fail CI with anything that is unexpected.

### PR DESCRIPTION
**Pull request by rossbar**
General maintenance to remove warnings from running the notebooks, including handling deprecated functions in scipy, which just released and rc for 1.12.

As noted in #154, failing on runtime warnings is not currently baked into CI. The way that I did this locally was to convert the tutorials into pure python files:

```bash
$ cd content
$ jupytext *.md --from myst --to py
```

Then run each of the python files with `-Werror`:

```bash
$ find content/ -name "*.py" -exec python -Werror {} \;
```

It's not *perfect*, but maybe something along these lines could be added to CI to resolve #154

### CODE REVIEW

File: content/tutorial-svd.md [hunk 1]
```diff
@@ -35,12 +35,12 @@ After this tutorial, you should be able to:
 
 ## Content
 
-In this tutorial, we will use a [matrix decomposition](https://en.wikipedia.org/wiki/Matrix_decomposition) from linear algebra, the Singular Value Decomposition, to generate a compressed approximation of an image. We'll use the `face` image from the [scipy.misc](https://docs.scipy.org/doc/scipy/reference/misc.html#module-scipy.misc) module:
+In this tutorial, we will use a [matrix decomposition](https://en.wikipedia.org/wiki/Matrix_decomposition) from linear algebra, the Singular Value Decomposition, to generate a compressed approximation of an image. We'll use the `face` image from the [scipy.datasets](https://docs.scipy.org/doc/scipy/reference/datasets.html) module:
 
 ```{code-cell}
-from scipy import misc
+from scipy import datasets
```
Comment (bsipocz) [inline content/tutorial-svd.md:41]: If we are committed to do SPEC0 type version support, then cannot yet do this change as `datasets` was added in 1.10 and we're still on 1.8 as oldest scipy

```suggestion
try:
    from scipy.datasets import face
except ImportError:
    from scipy.misc import face
```

File: content/tutorial-svd.md [hunk 2]
```diff
@@ -35,12 +35,12 @@ After this tutorial, you should be able to:
...
 
-img = misc.face()
+img = datasets.face()
```
Comment (bsipocz) [inline content/tutorial-svd.md:43]: In additon to the previous change, GH didn't let me do only one suggestion due to the deleted line in the diff
```suggestion
img =face()
```

**Review by bsipocz — CHANGES_REQUESTED** [review]
We need to deal with the oldest supported scipy version, otherwise this looks good to me.

**Comment by rossbar** [comment]
> We need to deal with the oldest supported scipy version, otherwise this looks good to me.

Good point - though I think there's a broader discussion to be had here. IMO the most important thing is supporting the latest releases. I'm not sure there's as much value in supporting the oldest devs.

**Comment by bsipocz** [comment]
> I'm not sure there's as much value in supporting the oldest devs.

I'm thinking mostly about end users who may stuck on older systems without the ability (or in fact the need) to upgrade versions to try to run a tutorial. Not sure how much this is a real concern.

**Comment by rossbar** [comment]
> I'm thinking mostly about end users who may stuck on older systems without the ability (or in fact the need) to upgrade versions to try to run a tutorial. Not sure how much this is a real concern.

Yeah that's a good point, and it jives with some of the other unofficial policies that the tutorials have adopted; e.g. not using sphinx-specific features (like intersphinx) in the tutorials themselves so they're still 100% functional as standalone notebooks. There's definitely a wider discussion to be had :)

**Comment by bsipocz** [comment]
What about getting in the try/except to make the current CI setup happy, and then discussing what would be sensible to do down the road?

**Review by bsipocz — APPROVED** [review]
Now it all looks good, thank you!

Maybe open an issue about the minimum scipy version, etc policies, so we don't forget to actually have the discussion :)

File: content/x_y-squared.npz
[the remark this answers is not in the document]
Comment (rossbar) [reply content/x_y-squared.npz:1]: Yeah, I must've caught it in a `git commit -a`. The file is already tracked, but it definitely shouldn't be. I'll make sure to back out the "changes" to the file here. We should remove it entirely from being tracked in a separate PR IMO.

[the remark this answers is not in the document]
Comment (bsipocz) [reply content/x_y-squared.npz:1]: I got bitten early on with `git commit -a` when working with submodules that I never use it ever :)

**Comment by bsipocz** [comment]
Thanks @rossbar!

**Pull request merged**